internal/impl/mail/input_pop3.go

```diff
@@ -0,0 +1,299 @@
+	passwordParam          = "password"
+	tlsParam               = "tls"
+	tlsEnabledParam        = "enabled"
+	tlsSkipCertVerifyParam = "skip_cert_verify"
+	pullParam              = "pull"
+	countParam             = "count"
+	intervalParam          = "interval"
+	maxInFlightParam       = "max_in_flight"
+	timeoutParam           = "timeout"
+	maxBodySizeParam       = "max_body_size"
+	listPageSizeParam      = "list_page_size"
+)
+
+func pop3ConfigSpec() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Categories("Services").
+		Version("4.12.0").
+		Summary("Connects to a POP3 server and periodically reads the emails.").
+		Description(`
+Reads emails from a POP3 server. The emails are permanently deleted when delivered.
+
+### Metadata
+
+This input adds the following metadata fields to each message if present:
+
+` + "```" + `
+- Subject
+- From
+- To
+- Cc
+- Bcc
+- Date
+- Received
+- Content-Type
+- Return-Path
+- Delivered-To
+` + "```" + `
+You can access these metadata fields using [function interpolation](/docs/configuration/interpolation#bloblang-queries).
+`).
+		Field(service.NewStringField(hostParam).
+			Description("The host of the POP3 server").
+			Example("pop.gmail.com")).
+		Field(service.NewIntField(portParam).
+			Description("The port of the POP3 server.").
+			Default(995)).
+		Field(service.NewStringField(usernameParam).
+			Description("The username to login on the POP3 server.")).
+		Field(service.NewStringField(passwordParam).
+			Description("The password to login on the POP3 server.")).
+		Field(service.NewObjectField(tlsParam,
+			service.NewBoolField(tlsEnabledParam).
+				Description(`Whether TLS is enabled.`),
+			service.NewBoolField(tlsSkipCertVerifyParam).
+				Description(`Whether to skip server side certificate verification.`)).Advanced()).
+		Field(service.NewObjectField(pullParam,
+			service.NewIntField(countParam).
+				Description(`An optional number of pulls to fetch messages, if set above 0 the specified number of pulls is executed and then the input will shut down.`).
+				Default(0),
+			service.NewDurationField(intervalParam).
+				Description(`The time interval at which message pulls should be performed, expressed as a duration string.`).
+				Default("5m")).Advanced()).
+		Field(service.NewDurationField(timeoutParam).
+			Description("The timeout to establish the connection with the POP3 server.").
+			Default("30s").Advanced()).
+		Field(service.NewIntField(maxInFlightParam).
+			Description("The maximum number of messages to have in flight at a given time. Increase this to improve throughput.").
+			Default(1).Advanced()).
+		Field(service.NewIntField(maxBodySizeParam).
+			Description("The max number of bytes when reading the message body.").
+			Default(4096).Advanced()).
+		Field(service.NewIntField(listPageSizeParam).
+			Description("The max number of emails returned by the POP3 server. This limit may be configured on the server or may be imposed by the server software.").
```

Comment:
Does this mean the value must match what the server has configured? Or just that it's optional as the server could be specifying it instead?